### PR TITLE
fix: vjudge can't get latest problems

### PR DESCRIPTION
**Pull request by guke1024**
目前的逻辑中，除codeforces以外的vjudge在启动oj时均无法正确获取最新题目，该方案给vjudge添加page字段解决此问题

### CODE REVIEW

**Comment by guke1024** [comment]
I have read the CLA Document and I hereby sign the CLA

File: packages/vjudge/src/model.ts [hunk 1]
```diff
@@ -55,6 +55,11 @@ class Service {
 
     async sync(domainId: string, resync = false, list: string) {
         let page = 1;
+        if (domainId !== 'codeforces') {
+            const oj = await coll.findOne({ type: domainId });
+            page = oj.page;
```
Comment (pandadtdyy) [inline packages/vjudge/src/model.ts:60]: L57 show that Hydro default sync from first page. You should deal with the first sync without page.

**Review by pandadtdyy — CHANGES_REQUESTED** [review]

**Comment by pandadtdyy** [comment]
The problem isn’t can not get, is your network problem.

File: packages/vjudge/src/model.ts [hunk 2]
```diff
@@ -55,6 +55,11 @@ class Service {
 
     async sync(domainId: string, resync = false, list: string) {
         let page = 1;
+        if (domainId !== 'codeforces') {
```
Comment (undefined-moe) [inline packages/vjudge/src/model.ts:58]: this **shouldn't** be hardcoded into model.
what if some other endpoints also add problem at the first page? (or middle?)
at current, if resyncing check for every page is required, emitting a problem list for **each** page works.
(although its not recommented, the update check runs every time after startup.)

File: packages/vjudge/src/model.ts [hunk 3]
```diff
@@ -87,6 +92,10 @@ class Service {
             page++;
             pids = await this.api.listProblem(page, resync, list);
         }
+        page--;
+        if (domainId !== 'codeforces') {
+            coll.updateMany({ type: domainId }, { $set: { page } });
```
Comment (undefined-moe) [inline packages/vjudge/src/model.ts:97]: index by domainId is incorrect

**Review by undefined-moe — CHANGES_REQUESTED** [review]

File: packages/vjudge/src/model.ts (hunk 2 above)
Comment (undefined-moe) [reply packages/vjudge/src/model.ts:58]: Maybe comparing problem count before syncing all page is a better solution.
Whether how adapter implements the optional `countProblem(list: string): Promise<number>` function depends on platform. Direct count API, count by page, fetch last page and get element count, etc.

**Pull request closed without merging**